modules/apigee_edge_apidocs/src/Entity/ApiDoc.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_apidocs\Entity;
+
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityChangedTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+
+/**
+ * Defines the API Doc entity.
+ *
+ * @ingroup apigee_edge_apidocs
+ *
+ * @ContentEntityType(
+ *   id = "apidoc",
```

Comment:
I am wondering whether it would be sensible to prefix the entity type ID by apigee, so 'apigee_apidocs'. This could avoid some potential conflict, even I guess the risk would be quite low. Sadly this would require a lot of changes all over the place.

Reply:
I created another task for that: https://apigeesc.atlassian.net/browse/DRUP-619